crates/sys/src/canvas.rs

```diff
@@ -1,12 +1,116 @@
 //! Low-level bindings to the Canvas API.
 
 use core::ffi::c_char;
+use core::fmt::Display;
 
 use crate::opaque;
 
 opaque!(Canvas);
 
+#[repr(C)]
```

Comment:
Use [`#[repr(transparent)]`](https://doc.rust-lang.org/reference/type-layout.html#the-transparent-representation) when creating C-compatible "new-types".

The issue with using `#[repr(C)]` in this specific case is that `Align` will have the layout of a C `struct` with a single `uint8_t`, as opposed to the layout of a bare `uint8_t`.

Reply:
In this case it would also be OK to use an `enum` (with `#[repr(C)]`).

The reason for creating a new-type for `Status` was that that it's undefined behaviour to cast an unknown `int` to a Rust `enum` and we have no control over what the C API might return. However when it's the other way around, we know the Rust values are always valid to pass to the C API.

Reply:
Done!

> The reason for creating a new-type for Status was that that it's undefined behaviour to cast an unknown int to a Rust enum and we have no control over what the C API might return.

My usual approach for such enums is to implement `From<u32>`, but the downside is you have to implement the mappings from value to enum manually.